src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -544,6 +546,11 @@ public static void approveSubmission(
         SubmissionDao submissionDao = DaoService.getSubmissionDao();
         Submission withheldSubmission = submissionDao.getFirstPassingSubmission(studentNetId, phase);
 
+        // Check that this phase hasn't already been approved
+        if (withheldSubmission.verifiedStatus() != Submission.VerifiedStatus.Unapproved) {
+            throw new IllegalArgumentException(studentNetId + " needs no approval for phase " + phase);
```

Comment:
Do you think a `RuntimeException` would be a better kind of exception to throw here?

Reply:
Fair enough